### ISSUE TITLE
Is it correct that the Trivy Server Mode Vulnerability DB update is normal?

### ISSUE THREAD

**Issue by happylie** [aquasecurity/trivy#2187]
Is it correct that the current Trivy Server Mode Vulnerability DB update is normal?

The UpdatedAt/NextUpdate date will not change after the time you ran.
I would appreciate it if you can check if it is being updated normally.

```
$ trivy -v
Version: 0.27.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-05-12 06:06:17.231057123 +0000 UTC
  NextUpdate: 2022-05-12 12:06:17.231056623 +0000 UTC
  DownloadedAt: 2022-05-26 23:35:53.026928539 +0000 UTC


$ ls -alth
-rw-r--r-- 1 test test  153  5 27 08:35 metadata.json
-rw------- 1 test test 298M  5 27 08:35 trivy.db
```

**Comment by afdesk** [comment]
@happylie yes, your dates look a little strange, it seems Trivy didn't update the database. 

could you update the db without scanning?
```sh
$ trivy image --download-db-only
```

**Comment by happylie** [comment]
@afdesk 
If you talk about the current situation a little more, you are running in server mode and running and running as follows.
```
$ trivy server --cache-backend redis://127.0.0.1:6379 --listen 127.0.0.1:4954 >> "trivy.log" &

$ ps -ef | grep trivy
test  7663     1  1 10:33 pts/1    00:00:02 trivy server --cache-backend redis://127.0.0.1:6379 --listen 127.0.0.1:4954
```

And we used the "-download-db-only" option, but nothing is changed.

When you run a new one, you can see that it applies normally as UpdatedAt/NextUpdate date as below.
```
$ trivy -v
Version: 0.27.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-05-29 00:11:41.445266375 +0000 UTC
  NextUpdate: 2022-05-29 06:11:41.445266075 +0000 UTC
  DownloadedAt: 2022-05-29 01:33:26.146409354 +0000 UTC
```

**Comment by afdesk** [comment]
Could you run `trivy` in debug mode?
```sh
$ trivy --debug server --cache-backend redis://127.0.0.1:6379 --listen 127.0.0.1:4954
```

**Comment by afdesk** [comment]
if Trivy works without the Internet you'll see `ERROR` in the logs:
```sh
2022-06-01T18:50:11.252+0600    ERROR   failed DB hot update:
....
```

**Comment by happylie** [comment]
I will run it again with Debug Mode and talk to you.

Previously, existing Log files are being updated normally as below.
```
$ trivy -v
Version: 0.27.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-05-29 06:06:15.432519897 +0000 UTC
  NextUpdate: 2022-05-29 12:06:15.432519597 +0000 UTC
  DownloadedAt: 2022-06-01 23:40:50.894367579 +0000 UTC


$ cat trivy.log
2022-06-02T05:40:32.691+0900    Updating DB...
2022-06-02T05:40:36.353+0900    Suspending all requests during DB update
2022-06-02T05:40:36.353+0900    Waiting for all requests to be processed before DB update...
2022-06-02T05:40:37.190+0900    Reopening DB...
2022-06-02T06:40:37.267+0900    Updating DB...
2022-06-02T06:40:42.422+0900    Suspending all requests during DB update
2022-06-02T06:40:42.423+0900    Waiting for all requests to be processed before DB update...
2022-06-02T06:40:43.043+0900    Reopening DB...
2022-06-02T07:40:43.232+0900    Updating DB...
2022-06-02T07:40:46.948+0900    Suspending all requests during DB update
2022-06-02T07:40:46.948+0900    Waiting for all requests to be processed before DB update...
2022-06-02T07:40:47.608+0900    Reopening DB...
```

**Comment by afdesk** [comment]
also could you use the latest trivy? 
thanks!

**Comment by afdesk** [comment]
@happylie it'll be useful. thanks

**Comment by happylie** [comment]
@afdesk 
I think the latest version is the same.
Is there anything wrong with me?
Authority or permission?
```
$ trivy -v
Version: 0.28.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-06-03 06:06:41.981063556 +0000 UTC
  NextUpdate: 2022-06-03 12:06:41.981062956 +0000 UTC
  DownloadedAt: 2022-06-03 20:46:00.102482755 +0000 UTC


$ ps -ef | grep trivy
test 12158     1  0  6 03 ?      00:01:01 trivy --debug server --cache-backend redis://127.0.0.1:6379 --listen 127.0.0.1:4954


$ ls -alth /home/test/.cache/trivy/db
-rw-r--r-- 1 test test  153  6  4 05:46 metadata.json
-rw------- 1 test test 297M  6  4 05:46 trivy.db


$ cat trivy.log
$ cat trivy.log
2022-06-03T09:44:47.130+0900  INFO  Redis cache: redis://127.0.0.1:6379
2022-06-03T09:44:47.130+0900  DEBUG cache dir:  /home/test/.cache/trivy
2022-06-03T09:44:47.130+0900  DEBUG There is no valid metadata file: unable to open a file: open /home/test/.cache/trivy/db/metadata.json: no such file or directory
2022-06-03T09:44:47.130+0900  INFO  Need to update DB
2022-06-03T09:44:47.130+0900  INFO  DB Repository: ghcr.io/aquasecurity/trivy-db
2022-06-03T09:44:47.130+0900  INFO  Downloading DB...
2022-06-03T09:44:47.130+0900  DEBUG no metadata file
2022-06-03T09:44:52.588+0900  DEBUG Updating database metadata...
2022-06-03T09:44:52.588+0900  DEBUG DB Schema: 2, UpdatedAt: 2022-06-03 00:10:07.878823411 +0000 UTC, NextUpdate: 2022-06-03 06:10:07.878823111 +0000 UTC, DownloadedAt: 2022-06-03 00:44:52.588693958 +0000 UTC
2022-06-03T09:44:52.589+0900  INFO  Listening 127.0.0.1:4954...
2022-06-03T10:44:52.589+0900  DEBUG Check for DB update...
2022-06-03T10:44:52.590+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T11:44:52.592+0900  DEBUG Check for DB update...
2022-06-03T11:44:52.593+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T12:44:52.594+0900  DEBUG Check for DB update...
2022-06-03T12:44:52.595+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T13:44:52.596+0900  DEBUG Check for DB update...
2022-06-03T13:44:52.597+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T14:44:52.598+0900  DEBUG Check for DB update...
2022-06-03T14:44:52.599+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T15:44:52.600+0900  DEBUG Check for DB update...
2022-06-03T15:44:52.601+0900  INFO  Updating DB...
2022-06-03T15:44:57.332+0900  DEBUG Updating database metadata...
2022-06-03T15:44:57.332+0900  INFO  Suspending all requests during DB update
2022-06-03T15:44:57.332+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-03T15:44:58.039+0900  INFO  Reopening DB...
2022-06-03T16:44:58.370+0900  DEBUG Check for DB update...
2022-06-03T16:44:58.371+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T17:44:58.373+0900  DEBUG Check for DB update...
2022-06-03T17:44:58.374+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T18:44:58.375+0900  DEBUG Check for DB update...
2022-06-03T18:44:58.376+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T19:44:58.376+0900  DEBUG Check for DB update...
2022-06-03T19:44:58.377+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T20:44:58.378+0900  DEBUG Check for DB update...
2022-06-03T20:44:58.379+0900  DEBUG DB update was skipped because the local DB is the latest
2022-06-03T21:44:58.380+0900  DEBUG Check for DB update...
2022-06-03T21:44:58.381+0900  INFO  Updating DB...
2022-06-03T21:45:17.076+0900  DEBUG Updating database metadata...
2022-06-03T21:45:17.076+0900  INFO  Suspending all requests during DB update
2022-06-03T21:45:17.076+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-03T21:45:17.763+0900  INFO  Reopening DB...
2022-06-03T22:45:17.992+0900  DEBUG Check for DB update...
2022-06-03T22:45:17.993+0900  INFO  Updating DB...
2022-06-03T22:45:22.601+0900  DEBUG Updating database metadata...
2022-06-03T22:45:22.601+0900  INFO  Suspending all requests during DB update
2022-06-03T22:45:22.601+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-03T22:45:23.637+0900  INFO  Reopening DB...
2022-06-03T23:45:23.717+0900  DEBUG Check for DB update...
2022-06-03T23:45:23.718+0900  INFO  Updating DB...
2022-06-03T23:45:28.667+0900  DEBUG Updating database metadata...
2022-06-03T23:45:28.668+0900  INFO  Suspending all requests during DB update
2022-06-03T23:45:28.668+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-03T23:45:29.291+0900  INFO  Reopening DB...
2022-06-04T00:45:29.713+0900  DEBUG Check for DB update...
2022-06-04T00:45:29.714+0900  INFO  Updating DB...
2022-06-04T00:45:33.266+0900  DEBUG Updating database metadata...
2022-06-04T00:45:33.267+0900  INFO  Suspending all requests during DB update
2022-06-04T00:45:33.267+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-04T00:45:33.932+0900  INFO  Reopening DB...
2022-06-04T01:45:34.065+0900  DEBUG Check for DB update...
2022-06-04T01:45:34.066+0900  INFO  Updating DB...
2022-06-04T01:45:38.709+0900  DEBUG Updating database metadata...
2022-06-04T01:45:38.709+0900  INFO  Suspending all requests during DB update
2022-06-04T01:45:38.709+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-04T01:45:39.401+0900  INFO  Reopening DB...
2022-06-04T02:45:39.699+0900  DEBUG Check for DB update...
2022-06-04T02:45:39.700+0900  INFO  Updating DB...
2022-06-04T02:45:44.271+0900  DEBUG Updating database metadata...
2022-06-04T02:45:44.271+0900  INFO  Suspending all requests during DB update
2022-06-04T02:45:44.271+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-04T02:45:45.417+0900  INFO  Reopening DB...
2022-06-04T03:45:45.503+0900  DEBUG Check for DB update...
2022-06-04T03:45:45.504+0900  INFO  Updating DB...
2022-06-04T03:45:49.641+0900  DEBUG Updating database metadata...
2022-06-04T03:45:49.642+0900  INFO  Suspending all requests during DB update
2022-06-04T03:45:49.642+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-04T03:45:50.826+0900  INFO  Reopening DB...
2022-06-04T04:45:50.899+0900  DEBUG Check for DB update...
2022-06-04T04:45:50.900+0900  INFO  Updating DB...
2022-06-04T04:45:54.474+0900  DEBUG Updating database metadata...
2022-06-04T04:45:54.474+0900  INFO  Suspending all requests during DB update
2022-06-04T04:45:54.474+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-04T04:45:55.074+0900  INFO  Reopening DB...
2022-06-04T05:45:55.472+0900  DEBUG Check for DB update...
2022-06-04T05:45:55.474+0900  INFO  Updating DB...
2022-06-04T05:46:00.102+0900  DEBUG Updating database metadata...
2022-06-04T05:46:00.102+0900  INFO  Suspending all requests during DB update
2022-06-04T05:46:00.102+0900  INFO  Waiting for all requests to be processed before DB update...
2022-06-04T05:46:00.834+0900  INFO  Reopening DB...
```

**Comment by afdesk** [comment]
it's a bit strange. your logs look normal, it seems that the database updates correctly, but `metadata.json` contains wrong dates...
could you take a look at ` `metadata.json`? 
```sh
$ cat /home/test/.cache/trivy/db/metadata.json | jq
```

give me some time to investigate it more.

**Comment by happylie** [comment]
@afdesk 
This is what I have checked now.
```
$ date -u
2022. 06. 07. 00:39:41 UTC

$ trivy -v
Version: 0.28.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-06-03 06:06:41.981063556 +0000 UTC
  NextUpdate: 2022-06-03 12:06:41.981062956 +0000 UTC
  DownloadedAt: 2022-06-06 23:52:49.227046925 +0000 UTC

$ cat /home/test/.cache/trivy/db/metadata.json
{"Version":2,"NextUpdate":"2022-06-03T12:06:41.981062956Z","UpdatedAt":"2022-06-03T06:06:41.981063556Z","DownloadedAt":"2022-06-06T23:52:49.227046925Z"}
```

Trivy is being put to good use.
Even if it takes time, I hope that the content will be corrected, and if there is any necessary information, I will support you at any time.

**Comment by afdesk** [comment]
@happylie 
could you check the latest trivy (>= 0.29.0)?
there were several fixes in this update.
thanks!

**Comment by happylie** [comment]
@afdesk 
Yes, sir.
Now we've confirmed that 0.29.2 Version is the latest
I will install the version and share the relevant contents after the test.

( cc : @erikgb )
It would be good to refer to the above.

**Comment by afdesk** [comment]
@happylie took a look at this issue again.
it seems there is any issue with access to the db repository.
I could see the same behavior for the db repository from my custom regestry, but I can see that you use the default repo (ghcr.io/aquasecurity/trivy-db). 

maybe it's cached somewhere?

**Comment by happylie** [comment]
@afdesk 
After installing 0.29.2 version, the update part is still the same.
Also, we didn't find any cache anywhere in the middle.

I installed a new system today.
After 6 hours, I will check the update and share it with you again.

**Comment by afdesk** [comment]
@happylie thanks for your report. I'm taking a look more

**Comment by happylie** [comment]
@afdesk 
It still looks the same.
There is no part that is cached in the middle.
Am I doing something wrong?

```
$ date -u
2022. 07. 04. (mon) 00:04:12 UTC

$ trivy -v
Version: 0.29.2
Vulnerability DB:
  Version: 2
  UpdatedAt: 2022-06-27 12:07:10.959688329 +0000 UTC
  NextUpdate: 2022-06-27 18:07:10.959687429 +0000 UTC
  DownloadedAt: 2022-07-03 23:30:41.7074888 +0000 UTC

$ cat /home/test/.cache/trivy/db/metadata.json
{"Version":2,"NextUpdate":"2022-06-27T18:07:10.959687429Z","UpdatedAt":"2022-06-27T12:07:10.959688329Z","DownloadedAt":"2022-07-03T23:30:41.7074888Z"}
```

**Comment by afdesk** [comment]
@happylie thanks
it doesn't look like you're doing anything wrong

**Comment by happylie** [comment]
@afdesk 
Hi~ 

May I know if the current problem has been solved?
Thank you.

**Comment by afdesk** [comment]
Hi @happylie 
there are several issues with db updating in server mode.
but unfortunately, I can't reproduce them...
we're investigating it

**Comment by happylie** [comment]
@afdesk 
Yes, I got it.
Then I'll wait for it to be resolved.

Thank you.

**Comment by afdesk** [comment]
@happylie we found a reason. 
it fixed in `v0.30.3`.
thanks a lot for your reports

**Comment by happylie** [comment]
@afdesk 
Thank you.
We have confirmed that the contents will be updated normally.
